addons/website_forum/static/src/js/website_forum.js

```diff
@@ -34,78 +165,96 @@ odoo.define('website_forum.website_forum', function (require) {
             }
             $(ev.currentTarget).after($warning);
         }
-    });
-
-    $('.o_js_forum_tag_follow').hover(
-        function(event) {
+    },
+    /**
+     * @private
+     * @param {Object} ev
+     */
+    _onHoverTagFollowBox: function (ev) {
+        $(function (ev) {
             $(this).find('.o_forum_tag_follow_box').stop().fadeIn().css('display','block');
-        },
-        function(event) {
+        });
+        $(function (ev) {
             $(this).find('.o_forum_tag_follow_box').stop().fadeOut().css('display','none');
-    });
-
-    $('.o_forum_profile_pic_edit').on('click', function(ev) {
+        });
+    },
+    /**
+     * @private
+     * @param {Object} ev
+     */
+    _onEditProfilePic: function (ev) {
         ev.preventDefault();
         $(this).closest('form').find('.o_forum_file_upload').trigger('click');
-    });
-
-    $('.o_forum_file_upload').on('change', function() {
+    },
+    /**
+     * @private
+     * @param {Object} ev
+     */
+    _onChangeFileUpload: function (ev) {
         if (this.files.length) {
             var $form = $(this).closest('form');
             var reader = new window.FileReader();
-            reader.onload = function(ev) {
+            reader.onload = function (ev) {
                 $form.find('.o_forum_avatar_img').attr('src', ev.target.result);
             };
             reader.readAsDataURL(this.files[0]);
             $form.find('#forum_clear_image').remove();
         }
-    });
-
-    $('.o_forum_profile_pic_clear').click(function() {
+    },
+    /**
+     * @private
+     */
+    _onProfilePicClear: function () {
         var $form = $(this).closest('form');
         $form.find('.o_forum_avatar_img').attr("src", "/web/static/src/img/placeholder.png");
         $form.append($('<input/>', {
             name: 'clear_image',
             id: 'forum_clear_image',
             type: 'hidden',
         }));
-    });
-
-    // Extended user biography toogle
-    $('.o_forum_user_info').hover(
-        function () {
-           $(this).parent().find('.o_forum_user_bio_expand').delay(500).toggle('fast');
-        },
-        function () {
+    },
+    /**
+     * @private
+     */
+    _onHoverUserInfo: function () {
+        $(function () {
+           $(this).parent().find('.o_forum_user_bio_expand').delay(500).toggle('fast');            
+        });
+        $(function () {
             $(this).parent().find('.o_forum_user_bio_expand').clearQueue();
-        }
-    );
-
-    $('.o_forum_user_bio_expand').hover(
-        function () {},
-        function () {
+        });
```

Comment:
Here you write "on dom ready, execute these two functions"... not sure why... I think you did not understand the previous code.

Reply:
I guess I did not understand the previous code.